### PR TITLE
Added Documentation for Diameter of Binary Tree

### PR DESCRIPTION
**Pull request by Saatvik21**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

`Yes`

### Description

I've added an article for diameter of binary tree

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

Fixes: #442

### CODE REVIEW

**Comment by Saatvik21** [comment]
@HarshCasper @iamrajiv Please review this PR!

File: docs/competitive-programming/diameter-of-BT.mdx [hunk 1]
```diff
@@ -0,0 +1,294 @@
...
+
+Here, we will use recursive approach to solve this problem.
+
+* Check whether the node passed in recursive function is null; if yes, then return `0`.
+* Recursively find the height of left sub-tree and right sub-tree.
+* Recursively find the diameter of left sub-tree and right sub-tree.
+* Check whether the diameter passes through the root node
+  * If yes, then no. of nodes in the path will be `lh + rh + 1` where lh and rh are the heights of left sub-tree and right sub-tree respectively.
+  * Else, diameter will be `max(ld,rd)` where ld and rd are the diameters of left sub-tree and right sub-tree respectively.
+* Return the maximum value of above steps.
```
Comment (iamrajiv) [inline docs/competitive-programming/diameter-of-BT.mdx:39]:
```suggestion
* Check whether the node passed in recursive function is null; if yes, then return `0`.
* Recursively find the height of the left subtree and right sub-tree.
* Recursively find the diameter of the left sub-tree and right sub-tree.
* Check whether the diameter passes through the root node
  * If yes, then no. of nodes in the path will be `lh + rh + 1` where lh and rh are the heights of left sub-tree and right sub-tree respectively.
  * Else, the diameter will be `max(ld,rd)` where ld and rd are the diameters of the left subtree and right sub-tree respectively.
* Return the maximum value of the above steps.
```

**Review by iamrajiv — CHANGES_REQUESTED** [review]
@Saatvik21 Implement the suggested changes.

**Review by iamrajiv — APPROVED** [review]

**Review by HarshCasper — APPROVED** [review]

**Comment by HarshCasper** [comment]
Hi @Saatvik21 

Can you rebase and fix the build?

**Comment by Saatvik21** [comment]
@HarshCasper Although the build is successful but the merge conflict watcher is failing. Is there any problem? Also, should I rebase for my other PR as well #431

**Review by iamrajiv — APPROVED** [review]

**Comment by HarshCasper** [comment]
> @HarshCasper Although the build is successful but the merge conflict watcher is failing. Is there any problem?

Nope. You can ignore that.

**Pull request merged**